src/main/java/uk/gov/hmcts/reform/sscs/ccd/presubmit/writefinaldecision/WriteFinalDecisionAboutToStartHandler.java

```diff
@@ -0,0 +1,155 @@
+package uk.gov.hmcts.reform.sscs.ccd.presubmit.writefinaldecision;
+
+import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.isYes;
+
+import java.util.*;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.sscs.ccd.callback.Callback;
+import uk.gov.hmcts.reform.sscs.ccd.callback.CallbackType;
+import uk.gov.hmcts.reform.sscs.ccd.callback.PreSubmitCallbackResponse;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
+import uk.gov.hmcts.reform.sscs.ccd.presubmit.PreSubmitCallbackHandler;
+import uk.gov.hmcts.reform.sscs.util.SscsUtil;
+
+@Component
+@Slf4j
+@AllArgsConstructor
+public class WriteFinalDecisionAboutToStartHandler implements PreSubmitCallbackHandler<SscsCaseData> {
+    @Value("${feature.postHearings.enabled}")
+    private boolean isPostHearingsEnabled;
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, Callback<SscsCaseData> callback) {
+        return callbackType == CallbackType.ABOUT_TO_START
+                && callback.getEvent() == EventType.WRITE_FINAL_DECISION
+                && Objects.nonNull(callback.getCaseDetails())
+                && Objects.nonNull(callback.getCaseDetails().getCaseData());
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<SscsCaseData> handle(CallbackType callbackType, Callback<SscsCaseData> callback, String userAuthorisation) {
+        if (!canHandle(callbackType, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        CaseDetails<SscsCaseData> caseDetails = callback.getCaseDetails();
+        SscsCaseData sscsCaseData = caseDetails.getCaseData();
+
+        SscsUtil.setCorrectionInProgress(caseDetails, isPostHearingsEnabled);
+
+        if (!isPostHearingsEnabled
+                || !isYes(sscsCaseData.getPostHearing().getCorrection().getCorrectionFinalDecisionInProgress())) {
```

Comment:
isNoOrNull, or if you specifically want NO, then use NO.equals(...)

Reply:
done